### DISCUSSION TITLE
Transform Filter data by values: add Is/Is Not undefined matcher or detect undefined as null

### DISCUSSION THREAD

**Discussion by douglaswth** [grafana/grafana#37664] [General]
I am currently working on building a dashboard with panels that need to graph data calculated from two different datasources ("Mixed") and I am using the "Labels to fields" Transform and its built in Merge functionality to combine frames from both datasources, but sometimes there will be rows from either datasource that do not match up to the other. After noticing this was happening, I tried using the "Filter data by values" transform with a couple "Is null" match conditions for fields that come from one datasource or the other, but after applying that, I noticed that it was not working as expected.

After digging through the code, I came upon #30808, which seems to indicate that the Merge functionality is intentionally setting those fields to `undefined` rather than `null` which means the `===` used in the "Is null" does not actually match.

I'm wondering whether adding a new "Is undefined" matcher or just having the "Is null" matcher use `==` instead would be the right solution?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
